src/fancy-tools/freehandTool.js

```diff
@@ -0,0 +1,1143 @@
+          drawLinkedTextBox(context, element, data.handles.textBox, text,
+            data.handles, textBoxAnchorPoints, color, lineWidth, 0, true);
+        }
+      });
+    }
+
+    function textBoxText (data) {
+      const { meanStdDev, meanStdDevSUV, area } = data;
+      // Define an array to store the rows of text for the textbox
+      const textLines = [];
+
+      // If the mean and standard deviation values are present, display them
+      if (meanStdDev && meanStdDev.mean !== undefined) {
+        // If the modality is CT, add HU to denote Hounsfield Units
+        let moSuffix = '';
+
+        if (modality === 'CT') {
+          moSuffix = ' HU';
+        }
+
+        // Create a line of text to display the mean and any units that were specified (i.e. HU)
+        let meanText = `Mean: ${this.constructor._numberWithCommas(meanStdDev.mean.toFixed(2))}${moSuffix}`;
+        // Create a line of text to display the standard deviation and any units that were specified (i.e. HU)
+        let stdDevText = `StdDev: ${this.constructor._numberWithCommas(meanStdDev.stdDev.toFixed(2))}${moSuffix}`;
+
+        // If this image has SUV values to display, concatenate them to the text line
+        if (meanStdDevSUV && meanStdDevSUV.mean !== undefined) {
+          const SUVtext = ' SUV: ';
+
+          meanText += SUVtext + this.constructor._numberWithCommas(meanStdDevSUV.mean.toFixed(2));
+          stdDevText += SUVtext + this.constructor._numberWithCommas(meanStdDevSUV.stdDev.toFixed(2));
+        }
+
+        // Add these text lines to the array to be displayed in the textbox
+        textLines.push(meanText);
+        textLines.push(stdDevText);
+      }
+
+      // If the area is a sane value, display it
+      if (area) {
+        // Determine the area suffix based on the pixel spacing in the image.
+        // If pixel spacing is present, use millimeters. Otherwise, use pixels.
+        // This uses Char code 178 for a superscript 2
+        let suffix = ` mm${String.fromCharCode(178)}`;
+
+        if (!image.rowPixelSpacing || !image.columnPixelSpacing) {
+          suffix = ` pixels${String.fromCharCode(178)}`;
+        }
+
+        // Create a line of text to display the area and its units
+        const areaText = `Area: ${this.constructor._numberWithCommas(area.toFixed(2))}${suffix}`;
+
+        // Add this text line to the array to be displayed in the textbox
+        textLines.push(areaText);
+      }
+
+      return textLines;
+    }
+
+    function textBoxAnchorPoints (handles) {
+      return handles;
+    }
+  }
+
+  // END Implementation of baseAnnotationTool methods //
+
+  /**
+  * Event handler for called by the mouseDownActivate event, if tool is active and
+  * the event is not caught by mouseDownCallback.
+  *
+  * @event
+  * @param {Object} evt - The event.
+  */
+  addNewMeasurement (evt, interactionType) {
+    if (interactionType === 'touch') {
+      console.warn('No touch controls implemented for freehandTool.');
```

Comment:
This should never be called, as the `touchEventDispatcher` will filter out tools that don't have `touch` in their list of supported interaction types

Reply:
Aha, that's nifty, I was just being cautious whilst implementing this, I'll remove it.